src/routes/api/api-router.ts

```diff
@@ -82,16 +84,27 @@ ApiRouter.post(
 		const offset = Number(req.body.offset) || 0;
 		// only resync installations whose "updatedAt" date is older than x seconds
 		const inactiveForSeconds = Number(req.body.inactiveForSeconds) || undefined;
+		// A date to start fetching commit history(main and branch) from.
+		let commitsFromDate = req.body.commitsFromDate;
```

Comment:
you're changing the type of the object here from unknown to date.  I'd probably just stick to Date to make sure there's no confusion.